### PR TITLE
Mandrill Merge Vars

### LINKED ISSUE #39: Add a function to easily add merge vars using the MandrillAdapter
The merge vars can be pretty handy but it's a pain to use them. Something like

```elixir
email |> put_merge_params(users, fn(user) -> %{first_name: user.first_name} end)
```

### PR DESCRIPTION
**Pull request by jbernardo95**
This PR resolves #39 

Adds a function to easily add  *Mandrill Merge Vars* to an email.

Example:

```elixir
email |> put_merge_vars(users, fn(user) -> %{first_name: user.first_name} end)
```

### CODE REVIEW

**Comment by jbernardo95** [comment]
@paulcsmith Can I get a review here ? 🙂

File: lib/bamboo/adapters/mandrill_helper.ex [hunk 1]
```diff
@@ -36,6 +36,59 @@ defmodule Bamboo.MandrillHelper do
...
+          vars: [
+            %{
+              "name": "full_name",
+              "content": "User 2"
+            }
+          ]
+        }
+      ])
+
+  ## Example
```
Comment (paulcsmith) [inline lib/bamboo/adapters/mandrill_helper.ex]: What do you think about moving the example above the "A convenience function for:"

File: test/lib/bamboo/adapters/mandrill_helper_test.exs [hunk 2]
```diff
@@ -9,6 +9,44 @@ defmodule Bamboo.MandrillHelperTest do
...
+        full_name: "User 1"
+      },
+      %{
+        email: "user2@example.com",
+        full_name: "User 2"
+      }
+    ]
+
+    email = new_email
+    |> MandrillHelper.put_merge_vars(users, fn(user) -> %{ full_name: user.full_name } end)
```
Comment (paulcsmith) [inline test/lib/bamboo/adapters/mandrill_helper_test.exs]: Could you indent this two spaces? Either that or put it on the same line and make the function multiline like this:

```elixir
email = MandrillHelper.put_merge_vars email, users, fn(user) ->
  %{full_name: user.full_name}
end
```

**Comment by paulcsmith** [comment]
@jbernardo95 Thanks for the PR. This looks awesome! Just a couple comments

File: test/lib/bamboo/adapters/mandrill_helper_test.exs (hunk 2 above)
Comment (jbernardo95) [inline test/lib/bamboo/adapters/mandrill_helper_test.exs]: Still trying to get used to the code standards x)

Fixed.

**Comment by jbernardo95** [comment]
@paulcsmith Updated.

**Comment by paulcsmith** [comment]
Thanks @jbernardo95! Nice work

**Pull request merged**